File: 0538-Convert-BST-to-Greater-Tree/Java/Solution.java
public class Solution {
    
    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode() {}

        TreeNode(int val) {
            this.val = val;
        }

        TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    }

    private int sum = 0;

    /**
     * https://leetcode.cn/problems/convert-bst-to-greater-tree/
     * 
     * @param root
     * @return
     */
    public TreeNode convertBST(TreeNode root) {
        inorder(root);
        return root;
    }

    private void inorder(TreeNode root) {
        if (root == null) return;
        inorder(root.right);
        sum += root.val;
        root.val = sum;
        inorder(root.left);
    }

}

File: 0538-Convert-BST-to-Greater-Tree/Python/Solution.py
from typing import Optional


# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


"""
https://leetcode.cn/problems/convert-bst-to-greater-tree/description/
"""
class Solution:
    sum = 0;

    def convertBST(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        self.inorder(root)
        return root

    def inorder(self, root: Optional[TreeNode]):
        if root is None:
            return
        self.inorder(root.right)
        self.sum += root.val
        root.val = self.sum
        self.inorder(root.left)